packages/jazz-tools/src/tools/implementation/zodSchema/runtimeConverters/schemaFieldToCoFieldDef.ts

```diff
@@ -49,9 +50,22 @@ export function schemaFieldToCoFieldDef(schema: SchemaField) {
   } else {
     if ("_zod" in schema) {
       const zodSchemaDef = schema._zod.def;
-      if (zodSchemaDef.type === "optional") {
+      if (
+        zodSchemaDef.type === "optional" ||
+        zodSchemaDef.type === "nullable"
+      ) {
         const inner = zodSchemaDef.innerType as ZodPrimitiveSchema;
-        return schemaFieldToCoFieldDef(inner);
+        const coFieldDef: any = schemaFieldToCoFieldDef(inner);
+        if (
+          zodSchemaDef.type === "nullable" &&
+          coFieldDef === coField.optional.Date
+        ) {
+          throw new Error("Nullable z.date() is not supported");
```

Comment:
If you think there's a way to support nullable date coFields let me know, but I think this is good enough